odoo/tools/osutil.py

```diff
@@ -54,6 +55,24 @@ def clean_filename(name, replacement=''):
         return "Untitled"
     return re.sub(r'[^\w_.()\[\] -]+', replacement, name).lstrip('.-') or "Untitled"
 
+def is_host_up(host, port, *, AF=socket.AF_INET, SOCK=socket.SOCK_STREAM):
+    """ Determine if the host is up by openning a IPv4/TCP socket.
+
+    :param str host: the host to connect to
+    :param int port: the port to connect to
+    :param int AF: address family, socket.AF_INET (ipv4) by default
+    :param int SOCK: socket type, socket.SOCK_STREAM (tcp) by default
+    """
+    sock = socket.socket(AF, SOCK)
+    try:
+        sock.connect((host, port))
```

Comment:
`socket.create_connection` seems like a good utility function for essentially that.